Junk4Dollars-app/API/QueryAPI.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  QueryAPI.swift
+//  Junk4Dollars-app
+//
+//  Created by John Riccardi on 11/13/19.
+//  Copyright © 2019 John Riccardi. All rights reserved.
+//
+
+import Foundation
+
+public struct ApiQueryError: Error {
+    public enum ErrorKind {
+        case
+            noData
+    }
+
+    public let kind: ErrorKind
+}
+
+public protocol APICallable {
+    func makeApiCall(endpoint: String, whatToDoWithResponseData: @escaping (Data?, URLResponse?, Error?) -> Void) -> Void
+}
+
+public class QueryAPI: APICallable {
+    public func makeApiCall(endpoint: String, whatToDoWithResponseData: @escaping (Data?, URLResponse?, Error?) -> Void) -> Void {
+        let url = URL(string: endpoint)!
+        let request: URLRequest = URLRequest(url: url)
+
+        let task = URLSession.shared.dataTask(with: request) {data, response, error in
+            if let error = error {
+                print("ERROR: \(error)")
+            }
+
+            guard let data = data else { return }
```

Comment:
What would happen if the server returns a 500 without a body, or if the connection cannot be established (there's an `error`).

Clients are usually interested in knowing when an API call ends regardless of the result.
Imagine that you start showing a spinner when you start the API call and you need to hide it when it ends.